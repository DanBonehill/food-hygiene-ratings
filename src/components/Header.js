import React from "react";
import {Link} from "react-router-dom";

export const Header = () => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <Link className="header__title" to="/dashboard">
                    <h1>Food Hygiene Ratings</h1>
                </Link>
            </div>
        </div>
    </header>
);
