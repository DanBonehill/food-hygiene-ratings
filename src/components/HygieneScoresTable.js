import React from 'react';
import {connect} from 'react-redux';

const HygieneScoresTable = ({scores, totalScoresCount}) => (
    <div>
        <tr>
            <th>Rating</th>
            <th>Percentage</th>
        </tr>
        {
            scores.filter((score) => score.value !== 0).map((score) => (
                <tr>
                    <td>{score.title}</td>
                    <td>{Math.round(score.value / totalScoresCount * 100)}%</td>
                </tr>
            ))
        }
    </div>
);

const mapStateToProps = (state) => ({
    scores: state.hygieneScores.calculatedScores,
    totalScoresCount: state.hygieneScores.allScores.length
});

export default connect(mapStateToProps)(HygieneScoresTable)