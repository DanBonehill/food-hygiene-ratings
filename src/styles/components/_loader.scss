.loader {
  align-items: center;
  display: flex;
  justify-content: center;
  margin: $l-size;
}
