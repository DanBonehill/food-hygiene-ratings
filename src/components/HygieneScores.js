import React from 'react';
import {connect} from 'react-redux';

import HygieneScoresTable from "./HygieneScoresTable";

export const HygieneScores = ({selectedAuthority, loading}) => {
    return (
        <div>
            {!loading &&
                <div>
                    <h1>{selectedAuthority}</h1>
                    <table>
                        {selectedAuthority &&
                        <HygieneScoresTable/>
                        }
                    </table>
                </div>
            }
        </div>
    )
};

const mapStateToProps = (state) => ({
    selectedAuthority: state.authorities.selectedAuthority,
    loading: state.authorities.loading
});

export default connect(mapStateToProps)(HygieneScores)