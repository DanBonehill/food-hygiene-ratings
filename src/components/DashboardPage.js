import React from "react";
import axios from 'axios';
import {connect} from 'react-redux';
import Select from 'react-select';

import {addAuthorities} from '../actions/authorities';
import HygieneScores from "./HygieneScores";
import Loader from './Loader';

export class DashboardPage extends React.Component {
    state = {
        options: [],
        hygieneScores: [],
        selectedAuthority: '',
        loading: false
    };
    componentDidMount() {
        axios.get("http://api.ratings.food.gov.uk/authorities/basic/", {
            headers: {'x-api-version': 2}
        }).then((res) => {
            const options = [];

            res.data.authorities.forEach((authority) => {
                options.push({value: authority.LocalAuthorityId, label: authority.Name})
            });

            this.props.addAuthorities(options);
        }).catch((err) => {
            console.log(err)
        })
    }
    handleChange = (selectedAuthority) => {
        const authorityId = selectedAuthority.value;
        this.setState({selectedAuthority: selectedAuthority.label, loading: true});

        axios.get("http://api.ratings.food.gov.uk/establishments", {
            headers: {'x-api-version': 2},
            params: {
                "localAuthorityId": authorityId,
                "pageNumber": 1,
                "pageSize": 10000
            }
        }).then((res) => {
            const hygieneScores = [];

            res.data.establishments.forEach((establishment) => {
                hygieneScores.push(establishment.RatingValue)
            });

            this.setState({hygieneScores, loading: false})
        })
    };
    render() {
        return (
            <div className="content-container">
                <h3>Select an authority</h3>
                <Select
                    options={this.props.authorities}
                    onChange={this.handleChange}
                />
                {this.state.loading &&
                    <Loader/>
                }
                <HygieneScores
                    hygieneScores={this.state.hygieneScores}
                    selectedAuthority={this.state.selectedAuthority}
                    loading={this.state.loading}
                />
            </div>
        )
    }
}
const mapStateToProps = (state) => ({
    authorities: state.authorities
});

const mapDispatchToProps = (dispatch) => ({
    addAuthorities: (authorities) => dispatch(addAuthorities(authorities))
});

export default connect(mapStateToProps, mapDispatchToProps)(DashboardPage)