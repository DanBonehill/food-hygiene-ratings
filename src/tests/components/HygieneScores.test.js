import React from 'react';
import {shallow} from 'enzyme';
import {HygieneScores} from '../../components/HygieneScores';

describe('HygieneScoresTable snapshot', () => {
    test('should render correctly when loading false', () => {
        const selectedAuthority = "Coventry";
        const loading = false;
        const wrapper = shallow(<HygieneScores
            selectedAuthority={selectedAuthority}
            loading={loading}
        />);
        expect(wrapper).toMatchSnapshot()
    });

    test('should render correctly when loading true', () => {
        const selectedAuthority = "Coventry";
        const loading = true;
        const wrapper = shallow(<HygieneScores
            selectedAuthority={selectedAuthority}
            loading={loading}
        />);
        expect(wrapper).toMatchSnapshot()
    })
});