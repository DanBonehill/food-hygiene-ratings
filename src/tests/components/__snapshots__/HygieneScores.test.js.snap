// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HygieneScoresTable snapshot should render correctly 1`] = `<div />`;

exports[`HygieneScoresTable snapshot should render correctly when loading false 1`] = `
<div>
  <div>
    <h1>
      Coventry
    </h1>
    <table>
      <Connect(HygieneScoresTable) />
    </table>
  </div>
</div>
`;

exports[`HygieneScoresTable snapshot should render correctly when loading true 1`] = `<div />`;
